### ISSUE TITLE
How to Query multiple collection in one AQL?

### ISSUE THREAD

**Issue by alexyzhou** [arangodb/arangodb#8711]
## My Environment

* __ArangoDB Version__:        
* __Storage Engine__:             
* __Deployment Mode__:       
* __Deployment Strategy__:   
* __Configuration__:               
* __Infrastructure__:               
* __Operating System__:        
* __Total RAM in your machine__:        
* __Disks in use__:        
* __Used Package__:              

## Component, Query & Data

__Affected feature__:

__AQL query (if applicable)__:

__AQL explain (if applicable)__:

__Dataset__:

__Size of your Dataset on disk__:

__Replication Factor & Number of Shards (Cluster only)__:

## Steps to reproduce

1. 
2. 
3. 

__Problem__:
I have a bunch of collections in ArangoDB and I want to query them in one go, e.g.

for n in [collection1, collection2]:
  filter n.type = 'A'
  return n

I tried the above query in ArangoDB but it didn't give me the correct answer.
So how can I achieve that?

__Expected result__:

**Comment by dothebart** [comment]
Hi, 
you can either [use joins on multiple collections](https://docs.arangodb.com/3.4/AQL/Examples/Join.html)
or use [subqueries on multiple collections](https://docs.arangodb.com/3.4/AQL/Examples/CombiningQueries.html)

**Comment by alexyzhou** [comment]
Thanks for your reply. However I don't want to use join, I just want to query across multiple collections (union them actually). How can I do that?

**Comment by dothebart** [comment]
https://docs.arangodb.com/3.4/AQL/Functions/Array.html#union ?

**Comment by alexyzhou** [comment]
Maybe I didn't make my question clearly. I want to query multiple collections at once, let's say collection1 and collection2 both have documents with type='A'. To find them all, I want to use just one AQL to get all the documents matching the conditions, instead of query them one by one.

In my real system I will have a bunch of collections (~20), and I don't want to query them one by one.

Thanks.

**Comment by dothebart** [comment]
there currently is no way to iterate over a list of collections in AQL. You have to specify them, and query them one by one.

**Comment by alexyzhou** [comment]
ok thanks. Is there any plan to add it to the roadmap?

**Comment by Simran-B** [comment]
You could also use ArangoSearch to search over multiple collections.

The problem I see with a syntax like this:
```
for n in [collection1, collection2]:
filter n.type = 'A'
return n
```
is that in order to utilize indices, it might need to be desugared into something like:

```
for n in collection1
filter n.type = 'A'
return n

UNION

for n in collection2
filter n.type = 'A'
return n
```

but that could be tricky to implement, especially because it could also occur in sub-queries. You might produce very long running queries, because one collection might have an index on the attribute but another may not. I agree that it would be convenient in some cases from a user point of view though.

**Comment by alexyzhou** [comment]
Got it, I will check ArangoSearch. Thanks

**Comment by maxkernbach** [comment]
Hi @alexyzhou,

did you already have a chance to check Arangosearch? Is it helpful in your use-case?

**Comment by maxkernbach** [comment]
@alexyzhou

Just sending a follow-up. Could you please reply to my latest message?

**Comment by PeterLappo** [comment]
This can be done with a union. The for loops can have filters
```
return union (
  for x in collection1
  return x,
  for y in collection2
  return y
 )
```

**Comment by frytoli** [comment]
Hi there - I'm interested in using Arango Search to search over multiple document collections, as is mentioned above. However, I'm having a hard time finding references that specifically show how to set that functionality up. Do you have any resources or guidance you could share?

I do see in the [ArangoSearch Tutorial](https://www.arangodb.com/learn/search/tutorial/) the linking of a graph collection to a view:
```
var v = db._createView("v_imdb", "arangosearch");
v.properties({ links: { imdb_vertices: link }}) // Where imdb_vertices is a graph collection
```

Could I link multiple document collections like this?
```
view.properties({ links: { collection1: link, collection2: link, collection3: link}})
```

Thanks in advance!

**Comment by Simran-B** [comment]
@frytoli Yes, you can add multiple collections to a View like in your example.

**Comment by frytoli** [comment]
Wonderful - thanks for the response!
